packages/medusa/src/loaders/pg-connection.ts

```diff
@@ -1,53 +1,39 @@
 import { asValue, AwilixContainer } from "awilix"
 import { ConfigModule } from "../types/global"
-import { ContainerRegistrationKeys } from "@medusajs/utils"
-import { ClientConfig } from "pg"
-import knex from "knex"
+import { ContainerRegistrationKeys, ModulesSdkUtils } from "@medusajs/utils"
 
 type Options = {
   configModule: ConfigModule
   container: AwilixContainer
 }
 
 export default async ({ container, configModule }: Options): Promise<void> => {
+  if (container.hasRegistration(ContainerRegistrationKeys.PG_CONNECTION)) {
+    return
+  }
+
+  // Share a knex connection to be consumed by the shared modules
   const connectionString = configModule.projectConfig.database_url
-  const database = configModule.projectConfig.database_database
   const extra: any = configModule.projectConfig.database_extra || {}
   const schema = configModule.projectConfig.database_schema || "public"
+  const idleTimeoutMillis = extra.idleTimeoutMillis ?? undefined // prevent null to be passed
+  const poolMax = extra.max
 
-  // Share a knex connection to be consumed by the shared modules
-  if (!container.hasRegistration(ContainerRegistrationKeys.PG_CONNECTION)) {
-    const config: ClientConfig = {
-      connectionString,
-      database,
-      ssl: extra?.ssl ?? false,
-      idle_in_transaction_session_timeout:
-        extra.idle_in_transaction_session_timeout ?? undefined, // prevent null to be passed
-    }
-
-    // TODO: see later if it is possible to use the same connection with multiple pool config using this shared connection across the modules
-    // const pgConnection = await new Client(config).connect() // or any other method to create your connection here
+  delete extra.max
+  delete extra.idleTimeoutMillis
```

Comment:
thought: This seems to be a leaking abstraction. Should this not be handled in the mikro util?

Reply:
the extra comes from the module config from medusa-config, that is why I re manipulate it here to pass it down to the funtion with the expected shape

Reply:
What I mean is, should we change the `database_extra` to be an object, pass it to the `ModulesSdkUtil.createPgConnection` and do the data shaping close to where it is used.

Does this make sense? 😅

Reply:
Actually nevermind. This is nitpicking, at this point :)

Reply:
I see what you mean, but i think the util define what it accept as the input and then the consumer prepare its object shape according to the expectation of the util and not the other way around. Or maybe i ve not properly understood what you mean 😅